doc/service-proxy.md

```diff
@@ -219,18 +240,30 @@ circuited by the service proxy.
 ## "Local" Internal and External Traffic Policy
 
 If a service has `.spec.internalTrafficPolicy` set to `"Local"`, then
-"internal" traffic to the service (that is, traffic which comes from a
-pod or node in the cluster) should only be redirected to endpoint IPs
-on the node that the connection originated from. (If there are no
-endpoint IPs on that node then internal connections to the service
-from that node should be rejected.)
+"internal" traffic to the service (that is, traffic to the ClusterIPs)
+should only be redirected to endpoint IPs on the node that the
+connection originated from. If there are no endpoint IPs on that node
+then internal connections to the service from that node should be
+dropped.
 
 If a service has `.spec.externalTrafficPolicy` set to `"Local"`, then
-"external" traffic to the service (that is, traffic which comes from a
-source outside the cluster) should only be redirected to endpoint IPs
-on the node where the connection first arrives in the cluster. (If
-there are no local endpoint IPs on that node then external connections
-to the service via that node should be dropped, _not_ rejected.)
+"external" traffic to the service (that is, traffic to NodePorts,
+ExternalIPs, or Load Balancer IPs) should only be redirected to
+endpoint IPs on the node where the connection first arrives in the
+cluster. If there are no local endpoint IPs on that node then external
+connections to the service via that node should be dropped.
+
+(For external traffic policy in particular, it is important that
+"wrongly-directed" traffic be _dropped_, not _rejected_, because there
+is a race condition between when the proxy stops accepting new
+connections for a service on a particular node versus when the load
+balancer _notices_ that the proxy has stopped accepting new
+connections. During that time, if the load balancer sends a connection
+to a "bad" node, we drop the packets so that the client will think
```

Comment:
the client will eventually timeout and receive an error, since the tcp stack retries with the same tuple is not likely the  intermediate devices will reroute hte packets to other node

```
tcpdump -i any -n tcp port 80
tcpdump: data link type LINUX_SLL2
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on any, link-type LINUX_SLL2 (Linux cooked v2), snapshot length 262144 bytes
14:28:07.117111 eth0  Out IP 192.168.8.2.58830 > 10.96.61.91.80: Flags [S], seq 3360558320, win 64240, options [mss 1460,sackOK,TS val 2346934370 ecr 0,nop,wscale 7], length 0
14:28:08.119044 eth0  Out IP 192.168.8.2.58830 > 10.96.61.91.80: Flags [S], seq 3360558320, win 64240, options [mss 1460,sackOK,TS val 2346935372 ecr 0,nop,wscale 7], length 0
14:28:10.135037 eth0  Out IP 192.168.8.2.58830 > 10.96.61.91.80: Flags [S], seq 3360558320, win 64240, options [mss 1460,sackOK,TS val 2346937388 ecr 0,nop,wscale 7], length 0
14:28:14.203047 eth0  Out IP 192.168.8.2.58830 > 10.96.61.91.80: Flags [S], seq 3360558320, win 64240, options [mss 1460,sackOK,TS val 2346941456 ecr 0,nop,wscale 7], length 0
```


the reject will indicate something is wrong and the client will be able to retry sooner, in both cases AFAIK the client will have a network error

Reply:
The dropping behavior has been intentional since the beginning (["KEP"](https://github.com/kubernetes/community/blob/f4379b3c775e31b38df46fb174cdafceceaaca33/contributors/design-proposals/external-lb-source-ip-preservation.md#external-traffic-blackholed-on-nodes-with-no-local-endpoints), [implementation](https://github.com/kubernetes/kubernetes/pull/29409/files#diff-2ccafd63d57233fd0697b9852514604ab880e29aeb073035765ad7541cad1b35R1168)). It's possible that we've been doing it wrong all along, I guess?